app/services/actions/prospective_student/registration.rb

```diff
@@ -89,7 +89,7 @@ def student # rubocop:disable Metrics/MethodLength
 
       def user
         @user ||= begin
-          User.find_by(id_number: id_number) || User.new(
+          ::User.find_by(id_number: id_number) || ::User.new(
```

Comment:
Bunun için `find_or_initialize_by` kullanabiliriz: https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_initialize_by

Reply:
@huseyin good catch!  Source şu:

```ruby
# File activerecord/lib/active_record/relation.rb, line 200
    def find_or_initialize_by(attributes, &block)
      find_by(attributes) || new(attributes, &block)
    end
```

Reply:
find_or_initialize_by burası için uygun değil. Değişmeyeceğinden emin olduğun durumlar için kullanabilirsin. Yani istance oluşturulurken farklı öznitelikler var ve find işlemide bu öz niteliklere göre aramay yapacağından her zaman doğru çalışmayacaktır.

Reply:
find_by ve new farklı argümanlar alıyor bu kodda, birebir karşılamıyor evet

Reply:
Ezbere yazıyorum ama,

```ruby
user = ::User.find_or_initialize_by(id_number: id_number) do |record|
    record.assign_attributes(
        email:                 email,
        password:              id_number,
        password_confirmation: id_number
    )
end
```

bu şekilde çalışabilir. Emin değilim.

Reply:
Bence find_or_initialize_by ya hep ya hiç bir metod.  Olduğu gibi yazılamayacaksa eskisi daha okunur, müdahale etmeyelim.

Reply:
Block'lu hali de çalışır bu arada. Kullanımını daha önce gördüm (gitlab'in source kodundaydı sanırım)